lib/p5.play.js

```diff
@@ -4338,4 +3909,1314 @@ p5.prototype._warn = function(message) {
+            .translate(p5.Vector.mult(this._center, -1))
+            .translate(c);
+          this._onTransformChanged();
+        }.bind(this)
+      },
+
+      /**
+       * The center of the collision shape in local-space - if this collider is
+       * owned by a sprite, the offset of the collider center from the sprite center.
+       * @property offset
+       * @type {p5.Vector}
+       */
+      'offset': {
+        enumerable: true,
+        get: function() {
+          return this._offset.copy();
+        }.bind(this),
+        set: function(o) {
+          this._localTransform
+            .translate(p5.Vector.mult(this._offset, -1))
+            .translate(o);
+          this._onTransformChanged();
+        }.bind(this)
+      },
+
+      /**
+       * The local-space rotation of the collider, in radians.
+       * @property rotation
+       * @type {number}
+       */
+      'rotation': {
+        enumerable: true,
+        get: function() {
+          return this._rotation;
+        }.bind(this),
+        set: function(r) {
+          this._localTransform
+            .clear()
+            .scale(this._scale)
+            .rotate(r)
+            .translate(this._offset);
+          this._onTransformChanged();
+        }.bind(this)
+      },
+
+      /**
+       * The local-space scale of the collider
+       * @property scale
+       * @type {p5.Vector}
+       */
+      'scale': {
+        enumerable: true,
+        get: function() {
+          return this._scale.copy();
+        }.bind(this),
+        set: function(s) {
+          this._localTransform
+            .clear()
+            .scale(s)
+            .rotate(this._rotation)
+            .translate(this._offset);
+          this._onTransformChanged();
+        }.bind(this)
+      }
+    });
+
+    this._onTransformChanged();
+  };
+
+  /**
+   * Update this collider based on the properties of a parent Sprite.
+   * Descendant classes should override this method to adopt the dimensions
+   * of the sprite if `getsDimensionsFromSprite` is true.
+   * @method updateFromSprite
+   * @param {Sprite} sprite
+   * @see p5.CollisionShape.prototype.getsDimensionsFromSprite
+   */
+  p5.CollisionShape.prototype.updateFromSprite = function(sprite) {
+    this.setParentTransform(sprite);
+  };
+
+  /**
+   * Update this collider's parent transform, which will in turn adjust its
+   * position, rotation and scale in world-space and recompute cached values
+   * if necessary.
+   * If a Sprite is passed as the 'parent' then a new transform will be computed
+   * from the sprite's position/rotation/scale and used.
+   * @method setParentTransform
+   * @param {p5.Transform2D|Sprite} parent
+   */
+  p5.CollisionShape.prototype.setParentTransform = function(parent) {
+    if (parent instanceof Sprite) {
+      this._parentTransform
+        .clear()
+        .scale(parent.scale) // TODO: ScaleX and ScaleY?
+        .rotate(radians(parent.rotation))
+        .translate(parent.position);
```

Comment:
I'm not sure I understand your question.  `itemPos` and `otherPos` are indices into the array - if they were equal, we'd be checking a vector against itself.

A plain-english way to read this would be "Remove this vector if it is parallel to any vector that appears after it in the array."